client/plots/matrix.controls.js

```diff
@@ -1293,6 +1294,10 @@ export class MatrixControls {
 					this.parent.chartType == 'hierCluster' &&
 					(g.type == 'hierCluster' || g.name == this.parent.config.settings.hierCluster?.termGroupName)
 						? 'expression'
```

Comment:
for a clustering group, shouldn't hardcode value to 'expression' (remember it works for non-exp data too), it should pass group.dataType. where to find dataType?

Reply:
It can be found in plot config (state)

<img width="1154" alt="Screenshot 2024-07-15 at 3 06 12 PM" src="https://github.com/user-attachments/assets/6aa12b01-1ca0-4349-819a-6351173774de">